mypy/main.py

```diff
@@ -255,7 +295,12 @@ def add_invertible_flag(flag: str,
     parser.add_argument('-V', '--version', action='version',
                         version='%(prog)s ' + __version__)
     parser.add_argument('--python-version', type=parse_version, metavar='x.y',
-                        help='use Python x.y')
+                        help='use Python x.y', dest='special-opts:python_version')
+    parser.add_argument('--python-executable', action='store',
+                        help="Python executable whose installed packages will be"
+                             " used in typechecking.", dest='special-opts:python_executable')
+    parser.add_argument('--no-site-packages', action='store_true',
+                        help="Do not search for PEP 561 packages in the package directory.")
```

Comment:
Is this actually respected if both `--python-version` and `--python-executable` are given?

Reply:
No. `--python-executable` is incompatible with `--no-site-packages` (and I need to add a check for this).

Reply:
I don't think it should be - it should be possible to do;

```
mypy --python-executable /bin/i-have-no-idea-what-python-this-is --no-site-packages
```
and have it infer the python version yet not use site-packages.

Reply:
I mean, I feel like the user should know what version of Python they are using, that is somewhat critical information. Then they should just set `--python-version`.

However, I could have it set `--python-executable` to `None` after inference if the `--no-site-packages` flag is passed, but that feels like a hack to support a tenuous use case at best.